snuba/cli/subscriptions_scheduler.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import signal
+from typing import Any, Optional
+
+import click
+from arroyo import configure_metrics
+from arroyo.processing import StreamProcessor
+
+from snuba import environment
+from snuba.environment import setup_logging, setup_sentry
+from snuba.subscriptions.scheduler_consumer import SchedulerBuilder, Tick
+from snuba.utils.metrics.wrapper import MetricsWrapper
+from snuba.utils.streams.metrics_adapter import StreamMetricsAdapter
+
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.option(
+    "--entity",
+    "entity_name",
+    default="events",
+    type=click.Choice(["events", "transactions"]),
+    help="The entity to target",
+)
+@click.option("--partitions", type=int)
```

Comment:
I don't think I entirely understand your comment. Are you talking about putting partition information into messages on the commit log topic... isn't that already there?

Reply:
The issue is the total number of partitions. 

Let's say you are building a watermark across partitions on a topic that has 64 partitions. You need to know how many partitions the topic has to ensure you saw a specific timestamp from all of them.
The commit log message tells you which partition the message concerns, but it does not tell you that it is partition n out of 64.

If we increase the number of partitions for the topic now, how would we seamlessly transition the scheduler possibly without loosing messages? 
Options:
- do nothing. As soon as you see a commit message for a partition higher than 64 you assume the number increased. Though potentially we could have miscalculated a handful of watermarks. Which potentially we could re-evaluate and fix after the fact since this happens rarely. So this will not be needed 
- Use the admin interface upon start up of the scheduler to find the current partition count and avoid the config. But we need to figure out how to deal with older messages that may be in the pipeline regarding the old number of partitions. That can create some additional delay.
- add the total number of partitions in the commit log. But this is only moving the problem somewhere else as the snuba consumer do not know the total number of partitions.
- other ideas ?

Reply:
I think the root of the problem is a bit separate to this. It is unclear to me whether Snuba should be managing the number of partitions or the number of partitions is to be detected by Snuba and used from there. I think we have code in Snuba like topic bootstrap (https://github.com/getsentry/snuba/blob/8435eff51de45aeb095e37b63330e14bc6d0f38c/snuba/cli/bootstrap.py#L89) that implies the former but you are suggesting we should be doing the latter. It is a bit hard to pick the best approach here without more clarity on that.